core/consensus/grandpa/impl/grandpa_impl.cpp

```diff
@@ -151,7 +153,8 @@ namespace kagome::consensus::grandpa {
         scheduler_,
         round_state);
 
-    new_round->end();
+    new_round->end();  // it is okay, because we do not want to actually execute
```

Comment:
Would it make sense to comment why we do not want this?

Reply:
it is explained in grandpa_impl.hpp file